packages/shared/src/hooks/useFeed.ts

```diff
@@ -152,7 +158,9 @@ export default function useFeed<T>(
             index,
           }));
 
-          if (isAdsQueryEnabled) {
+          if (settings.showAcquisitionForm) {
+            posts.splice(adSpot, 0, { type: 'userAcquisition' });
```

Comment:
Sweet!

Reply:
In this case why removing it it will do a full rerender of the feed now taking ad into account? (this is more for my understanding)

Reply:
Yes, unfortunately removing the ad will cause a full re-render with this approach. However most of the items in the array stay the same, so it should be a rerender of list, where all of the items (except one) do not get rerendered, since the items are the same reference. At least that's my understanding.

I think we had the same problem in the previous iteration of this PR - e.g. we had a children list, which we changed, so all of the feed container got rerendered. But all of the children except one were the same reference.

On trying it out locally, I didn't see any perf hiccups, but I didn't run any benchmarks or check if my assumptions above are correct.

Reply:
Ah cool, if it looks alright locally it's fine we can give it a go.